LindseyKCrittendon/AccrualWorld
# form working without user info drop down. error message when required…

## Description
Expenses in nav bar leads to the create form where the user has to put in certain information to submit the form.

## Details
I created an ExpenseCreateViewModel since the Expense form needed a drop down of all Expense types.  I then refactored the first create method by creating an instance of that view model and inserting the data from ExpenseTypes.  I inserted a value with an id = 0 and text that tells the user to select a deduction category in the drop down.  I created a range in the expense type model to only insert info into the database if the ExpenseTypeId > 0.  As a result, the user receives an error message telling him/her to select from the drop down if they attempt to submit the form without.

In the second create method in the ExpensesController, I performed ModelState.Remove() on the User and UserId, and linked the the user's information automatically.  I created another instance of the view model and redirected the user to the details view for that specific expense once the form is submitted.

In the razor view for Create Expenses, I refactored the view to show the new drop down list, took out the user drop down, and I changed the hyperlink at the bottom of the form to allow the user to view previously added expenses if so desired.

## Testing Instructions
-git add and git commit whatever you're working on on your branch
git checkout master to switch to your local master branch
git fetch --all 
git checkout lc-createExpenses to switch to branch

Update-database in the Package Manager Console, and run the app.

The nav bar should only display the logo and the login option.  Click on the login and input Admin@Admin.com 
password: Admin8*

The nav bar should now display 'Income', 'Expenses', 'Mileage', 'Receipts/Invoices'.  Click on the expenses option.  You will be lead to a form.  Try to submit the form without supplying required information.  The only information not required is the ImagePath.  You will receive an error message.  Put the required information in and click submit.  You will be routed to the details view for the expense you just created.  Click on the Expenses tab again to go to the create form, and click on the option at the bottom to view your previously added expenses in a list.  You should see a list of expenses.
